styx-common/src/main/java/com/spotify/styx/util/ParameterUtil.java

```diff
@@ -40,25 +40,54 @@
 import java.time.format.SignStyle;
 import java.time.temporal.ChronoField;
 import java.time.temporal.ChronoUnit;
-import java.time.temporal.TemporalUnit;
 import java.util.List;
-import javaslang.control.Either;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
  * Helpers for working with {@link com.spotify.styx.model.Workflow} parameter handling.
  */
 public final class ParameterUtil {
 
-  public static final String HOUR_PATTERN = "yyyy-MM-dd'T'HH";
-  public static final String DAY_PATTERN = "yyyy-MM-dd";
+  private static final Pattern YEAR_PATTERN = Pattern.compile("(\\d{4})(-\\d{2})?");
+  private static final Pattern YEAR_MONTH_PATTERN = Pattern.compile("(\\d{4})-(\\d{2})");
 
   private ParameterUtil() {
   }
 
   private static final int MIN_YEAR_WIDTH = 4;
   private static final int MAX_YEAR_WIDTH = 10;
+
+  private static final DateTimeFormatter DATE_HOUR_FORMAT = new DateTimeFormatterBuilder()
```

Comment:
This stack alike builder is not quite readable. Hopefully we will never need to change it.